### PR TITLE
Adding git clone to Dockerfile & details about packages

### PR DESCRIPTION
**Pull request by corey-nm**
Following commands were tested, meaning all commands in README are tested:
- `docker build --build-arg CUDA_VERSION=11.1 -t sparseml_docker .`
- `docker build --build-arg CUDA_VERSION=11.1 --build-arg GIT_CHECKOUT=main -t sparseml_docker .`
- `docker build --build-arg CUDA_VERSION=11.1 --build-arg GIT_CHECKOUT=v1.0.1 -t sparseml_docker .`
- `docker pull ghcr.io/neuralmagic/sparseml:1.0.1-ubuntu18.04-cu11.1`
- `docker run -it sparseml_docker`

Additionally, I tested that subsequent executions of the same command used the latest image checkpoint that they could.

### CODE REVIEW

**Comment by corey-nm** [comment]
If this looks good I'll do pretty much the same thing in deepsparse

File: docker/README.md [hunk 1]
```diff
@@ -7,11 +7,32 @@ image (with CUDA support). The image comes with a pre-installed `sparseml`, as w
...
+
+```bash
+docker pull ghcr.io/neuralmagic/sparseml:1.0.1-ubuntu18.04-cu11.1
+```
+
+## Extend
+If you would like to customize the docker image, you can use the pre-built images as a base in your own `Dockerfile`:
+
+```Dockerfile
+from ghcr.io/neuralmagic/sparseml:1.0.1-ubuntu18.04-cu11.1
```
Comment (bfineran) [inline docker/README.md:21]: need to add example of either running the pulled image or tagging so it is compatible with the run example below

**Review by bfineran — CHANGES_REQUESTED** [review]
dockerfile update lgtm. please add in PR description scope of what was tested (ie maintain previous functionality, test updated functionality, test new/old doc commands)

File: docker/README.md (hunk 1 above)
Comment (corey-nm) [reply docker/README.md:21]: @bfineran would including a `docker tag ghcr.io/... sparseml_docker` in the pull command be sufficient for this? Then the run commands in the run section could be used without updates I think

Comment (corey-nm) [reply docker/README.md:22]: Added the docker tag command.

**Review by KSGulin — APPROVED** [review]
Nice 🙂

**Review by bfineran — APPROVED** [review]

**Pull request merged**